### DISCUSSION TITLE
Building Flappy bird project

### DISCUSSION THREAD

**Discussion by Programmer-Faraj** [ZacharyPatten/dotnet-console-games#70] [Q&A]
Hello there,

I would like to know how to start building a project like Flappy bird for practicing purposes.

Thank you

Kindly,

**Comment by ZacharyPatten** [comment]
@Mohammed2021-coder So sorry for the late response. I have been very busy lately.

So, hopefully you noticed that the repository does include a console Flappy Bird clone, so one way to start could be to review that example.

Flappy Bird is a pretty simple game to code though. Generally you have a fixed speed at which the map scrolls with randomly generated obstacles that force you to be at specific heights, and the only input from the player is if they flap their wings to trigger to set an upwards velocity. There needs to be a downwards acceleration, so ever frame/update you need to reduce the birds vertical velocity by a set amount.

If you are not aware there is a C# discord server with lots of people who can help here: https://discord.gg/csharp. I am also a member on there, and feel free to ping me. I also have a discord server if interested here: https://discord.gg/c3r4E4zN. But keeping the conversation here is totally fine too.

Did you already create a project? If yes what type of project did you create? Console? WinForms? Web? etc. Also, feel free to share your code and what you are currently stuck on and I can offer specific help.

**Reply by Programmer-Faraj** [reply]
Hey, 

No, it's totally alright for me bro, no worry at all. Wow, you just gave a whole big idea regarding the project structure of building Flappy Bird, recently I gave up the idea of building this game since my direction is trying to practice building multi-platform apps. So that's why I'm trying to come up with ideas for building simple projects like Console application apps. I have already built a project although it's a simple one that doesn't contain any complexity, still it was a learning experience. By the way, I'm very thankful for mentioning the Discord servers, since it's the best and most rewarding method to learn how to code and build
 projects.

**Reply by Programmer-Faraj** [reply]
I've checked the projects folder in your Github and I've noticed that they're all consisted of games. I need to practice building projects. Do you have any OOP project idea for someone who has learned the basics? Something similar to this! https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/tutorials/classes
